core/chain-configs/src/updateable_config.rs

```diff
@@ -2,16 +2,15 @@ use crate::metrics;
 use chrono::{DateTime, Utc};
 use near_primitives::time::Clock;
 use near_primitives::types::BlockHeight;
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Serialize, Serializer};
 use std::fmt::Debug;
 use std::sync::{Arc, Mutex};
 
 /// A wrapper for a config value that can be updated while the node is running.
 /// When initializing sub-objects (e.g. `ShardsManager`), please make sure to
 /// pass this wrapper instead of passing a value from a single moment in time.
 /// See `expected_shutdown` for an example how to use it.
-/// TODO: custom implementation for Serialize and Deserialize s.t. only value is necessary(JIRA:ND-283)
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
```

Comment:
The derived implementation of `Deserialize` is not compatible with the custom `Serialize` implementation. Please consider removing the derived `Deserialize` implementation.

Reply:
I removed the derived deserializer for RpcClientConfigResponse, ClientConfig and MutableConfigValue. The /client_config is still working perfectly fine locally. Tested and confirmed serializer is needed [here](https://github.com/near/nearcore/blob/7c197d0f3b9ca14d37fcfc35a7020a72ea8265b8/chain/jsonrpc/src/lib.rs#L1449). 
Also removed derived Debug for RpcClientConfigRequest and RpcClientConfigResponse as Debug does not seem necessary.